x/blob/types/payforblob.go

```diff
@@ -1,31 +1,49 @@
 package types
 
 import (
+	"bytes"
 	"crypto/sha256"
-	"fmt"
+	math "math"
 
 	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	shares "github.com/celestiaorg/celestia-app/pkg/shares"
 	"github.com/celestiaorg/nmt"
-	sdkclient "github.com/cosmos/cosmos-sdk/client"
+	"github.com/celestiaorg/nmt/namespace"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	"github.com/cosmos/cosmos-sdk/types/tx/signing"
-	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
 	"github.com/tendermint/tendermint/crypto/merkle"
 	coretypes "github.com/tendermint/tendermint/types"
+	"golang.org/x/exp/constraints"
 
 	appshares "github.com/celestiaorg/celestia-app/pkg/shares"
 )
 
 const (
-	URLMsgWirePayForBlob = "/blob.MsgWirePayForBlob"
-	URLMsgPayForBlob     = "/blob.MsgPayForBlob"
-	ShareSize            = appconsts.ShareSize
-	SquareSize           = appconsts.MaxSquareSize
-	NamespaceIDSize      = appconsts.NamespaceSize
+	URLBLobTx        = "/blob.BlobTx"
+	URLMsgPayForBlob = "/blob.MsgPayForBlob"
+	ShareSize        = appconsts.ShareSize
+	SquareSize       = appconsts.MaxSquareSize
+	NamespaceIDSize  = appconsts.NamespaceSize
 )
 
 var _ sdk.Msg = &MsgPayForBlob{}
 
+func NewMsgPayForBlob(signer string, nid namespace.ID, blob []byte) (*MsgPayForBlob, error) {
+	commitment, err := CreateCommitment(nid, blob, appconsts.ShareVersionZero)
```

Comment:
Is `commitment` a fixed size? (like a hash)

Reply:
Yes `commitment` is a fixed size hash

Ref:
1. https://github.com/celestiaorg/celestia-app/blob/a82110a281bf9ee95a9bf9f0492e5d091371ff0b/x/blob/types/payforblob.go#L164
2. https://github.com/tendermint/tendermint/blob/ab4238a0e27acff1cb4fe41cc4b1dcebbd706961/crypto/merkle/tree.go#L9-L13

Reply:
Ok so we could assert a specific size rather than non-zero in `ValidateBasic`:

```
if len(msg.ShareCommitment) == 0 {
	return ErrEmptyShareCommitment
}
```